src/components/Slider/Slider.tsx

```diff
@@ -1,63 +1,64 @@
-import React, { useCallback, useEffect, useRef, useState } from 'react'
+import React, { useCallback, useEffect, useRef } from 'react'
 
+import { useAppDispatch, useAppSelector } from 'hooks/reduxHooks'
 import swipeLeft from 'img/svg/swipe-left.svg'
 import swipeRight from 'img/svg/swipe-right.svg'
+import { nextPicture, previousPicture } from 'store/counterSliderSlice'
+import { useGetItemsQuery } from 'store/services/slider'
+
+import { Loading } from 'components/Loading'
+
 import { SliderItem } from './SliderItem'
-import { useHttp } from 'hooks/http.hooks'
-import { IImgSlider } from 'interface/IImgSlider'
 
 import styles from './Slider.module.scss'
 
 export const Slider = () => {
-    const [pictures, setPictures] = useState<IImgSlider[]>([])
-    const [indexPicture, setIndexPicture] = useState(0)
+    const { indexPicture } = useAppSelector(state => state.countSlider)
 
-    const refContainer = useRef<HTMLDivElement>(null) 
+    const { data: sliderPictures, isLoading } = useGetItemsQuery()
 
-    const { request } = useHttp()
+    const dispatch = useAppDispatch()
 
-    const getImgForSlider = useCallback(async() => {
-        const data = await request({ url: 'slider' })
-        setPictures(data)
-    }, [request])
+    const refContainer = useRef<HTMLDivElement>(null)
 
-    useEffect(() => {
-        getImgForSlider()
-    }, [getImgForSlider])
+    const onClickNext = useCallback(() => {
+        dispatch(nextPicture(sliderPictures?.length))
+    }, [dispatch, sliderPictures?.length])
 
-    const nextPicture = () => {
-        setIndexPicture(preState => pictures.length-1 === preState ? 0 : preState + 1)
-    }
-
-    const previousPicture = () => {
-        setIndexPicture(preState => preState === 0 ? pictures.length-1 : preState - 1)
-    }
+    const onClickPrevious = useCallback(() => {
```

Comment:
const onClickPrevious = useCallback(() => dispatch(previousPicture(sliderPictures?.length)), [dispatch, sliderPictures?.length])

Reply:
the type of functions must be the same

Reply:
fixed